### PR TITLE
Style: Styling delivery app with Mathilde

### PR DESCRIPTION
**Pull request by ioaNikas**
- [x] New style for the header
- [x] Prettier delivery list
- [x] Prettier delivery form
- [x] Defined custom css variables and font-size/font-weight into style.scss

### CODE REVIEW

File: libs/material/src/lib/delivery/stakeholder-item/stakeholder-item.component.html [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   <mat-card-content>
-    <h2>Signee:</h2>
+    <h4>Signee</h4>
     <p>{{ stakeholder.organization.name }}</p>
   </mat-card-content>
-  <mat-card-footer *ngIf="!(hasStakeholderSigned$ | async)" fxLayout="row" fxLayoutAlign="space-evenly center" class="footerLocked">
-    <mat-icon>lock_open</mat-icon>
-    <span>The signature will appear here</span>
+  <mat-card-footer *ngIf="!(hasStakeholderSigned$ | async)" fxLayout="row" fxLayoutAlign="center center" class="footerLocked">
+      <span><mat-icon class="lock-icon">lock_open</mat-icon></span>
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/stakeholder-item/stakeholder-item.component.html]: You don't need a span around a mat-icon.

File: libs/material/src/lib/delivery/stakeholder-item/stakeholder-item.component.html [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+    <span>Your signature will appear here</span>
+  </mat-card-footer>
+  <mat-card-footer *ngIf="hasStakeholderSigned$ | async" fxLayout="row" fxLayoutAlign="center center" class="footerSigned">
+    <span><mat-icon svgIcon="signed" class="sign-icon"></mat-icon></span>
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/stakeholder-item/stakeholder-item.component.html]: You don't need a span around the mat-icon.

File: libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.html [hunk 3]
```diff
@@ -1,30 +1,30 @@
 <mat-card class="container mat-elevation-z6">
-  <mat-card-title>Edit your material</mat-card-title>
+  <mat-card-title><h4>Edit your material</h4></mat-card-title>
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.html]: You should not have to add h4 for mat-card-title. Why do you do that ?

File: libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.html [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+      <mat-form-field appearance="outline" class="inputs">
         <mat-label>Category</mat-label>
         <input class="category" formControlName="category" matInput placeholder="Category" />
       </mat-form-field>
     </div>
 
-    <div class="step" fxLayout="column" fxLayoutAlign="space-around center">
-      <mat-form-field *ngIf="hasStep" appearance="outline">
+    <div class="step" fxLayout="column" fxLayoutAlign="space-between end">
+      <mat-form-field *ngIf="hasStep else noStep" appearance="outline" class="mat-select-step">
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.html]: It misses a ";" before the "else" I think.

File: libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.scss [hunk 5]
```diff
@@ -1,11 +1,26 @@
 mat-card {
-  margin-bottom: 4%;
+  margin-bottom: 20px;
+  padding: 30px 50px 50px 50px;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.scss]: Best would be to do : `padding: 50px; padding-top: 30px`. I find it more readible.

File: libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.html [hunk 6]
```diff
@@ -21,12 +21,19 @@
...
-      {{ material.step.name }}
-    </mat-card-title>
+  <mat-card-content
+    *ngIf="material.step"
+    class="content-step"
+    fxLayout="column"
+    fxLayoutAlign="start end"
+  >
+    <div>
+      <mat-card-subtitle class="step">
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.html]: I think this is a missuse of `mat-card-subtitle` and title. Just use normal html sementic here (you're inside the <mat-card-content>

File: libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss [hunk 7]
```diff
@@ -1,24 +1,41 @@
...
+.description-text {
+  font-size: 11px;
+  font-weight: 400;
+  font-style: italic;
+  margin-bottom: 0;
+}
+
+mat-card-actions {
+  visibility: hidden;
+  margin: 0 0 0 -6px;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss]: Divide in two : `margin: 0; margin-left: -6px`. And you're still using a LOT of negative values...

File: libs/material/src/lib/template/template-picker/template-picker.component.scss [hunk 8]
```diff
@@ -36,19 +48,18 @@ mat-card {
...
 }
 
 p {
-  color: rgba(0, 0, 0, 0.514);
-}
-
-h2 {
-  font-weight: bold;
+  font-size: 14px;
+  margin-top: -18px;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/template/template-picker/template-picker.component.scss]: negative.

File: libs/movie/src/lib/movie/view/view.component.ts [hunk 9]
```diff
@@ -33,8 +33,8 @@ export class ViewComponent implements OnInit, OnDestroy {
   public getStaticBySlug (scope: string, slug: string) {
     return getLabelBySlug (scope, slug) as string;
   }
-  
+
   ngOnDestroy() {
-    this.store.setActive(null);
+    // this.store.setActive(null);
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/view/view.component.ts]: Remove it if you don't need it.

File: libs/ui/src/lib/toolbar/header/header.component.html [hunk 10]
```diff
@@ -11,35 +9,53 @@
...
-    </a>
-  </mat-menu>
+  <form>
+    <mat-icon color="primary" class="search-icon">search</mat-icon>
+    <mat-form-field class="searchInput">
+      <input type="text" matInput placeholder="Search any film" />
+    </mat-form-field>
+  </form>
+
+  <ng-container>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/toolbar/header/header.component.html]: Why do you need a ng-container here ????

File: libs/ui/src/lib/toolbar/header/header.component.html [hunk 11]
```diff
@@ -11,35 +9,53 @@
...
+    <button mat-icon-button [matMenuTriggerFor]="notifications" class="notification-button">
+      <mat-icon
+        *ngIf="(notificationsLength$ | async) as length; else empty"
+        svgIcon="notifications"
+        matBadge="{{ length }}"
+        matBadgeColor="primary"
+        matBadgeSize="medium"
+        class="notification-icon"
+      ></mat-icon>
       <ng-template #empty>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/toolbar/header/header.component.html]: ng-template should be outside of the container, or it won't work....

File: libs/ui/src/lib/toolbar/header/header.component.scss [hunk 12]
```diff
@@ -37,3 +42,34 @@
...
+
+.notification-icon {
+  height: 25px;
+  width: 25px;
+}
+
+.account-icon {
+  height: 50px;
+  width: 50px;
+  margin-top: -5px;
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/toolbar/header/header.component.scss]: Negative

File: styles/style.scss [hunk 13]
```diff
@@ -70,7 +82,12 @@ mat-card {
 
 button {
   border-radius: 0px !important;
-  font-family: 'Montserrat', 'Helvetica Neue', sans-serif !important;
+  font-family: 'Montserrat', 'Helvetica Neue', sans-serif !important; // Only for menu in movie card, WHY?
+}
+
+button:disabled {
+  color: var(--disabled)!important;
```
Comment (GrandSchtroumpf) [inline styles/style.scss]: You should not need this !important. You should customize the mat-button component instead.

File: libs/material/src/lib/delivery/delivery-form/delivery-form.component.scss [hunk 14]
```diff
@@ -1,73 +1,96 @@
...
   box-shadow: 0px 2px 9px rgba(0, 0, 0, 0.2);
+  padding-bottom: 40px;
 }
 
 mat-sidenav-content {
   height: 100%;
   overflow: auto;
-  background-color: white;
-  padding-bottom: 5%;
+  background-color: #FFFFFF;
```
[the remark this answers is not in the document]
Comment (ioaNikas) [reply libs/material/src/lib/delivery/delivery-form/delivery-form.component.scss]: Should I create a css variable for white color to respect our logic ?
We override defaut background-color with white one, as default is `#FAFAFA`.

File: libs/material/src/lib/delivery/delivery-item/delivery-item.component.html [hunk 15]
```diff
@@ -4,27 +4,39 @@
     </span>
     <span class="second-dot mat-elevation-z4" [ngStyle]="{ 'background-color': getRandomColor() }">
     </span>
-    <span *ngFor="let stakeholder of delivery.stakeholders; last as isLast" class="names" fxLayout="row">
-      <h4>{{stakeholder.organization.name}}</h4>
+    <span
```
[the remark this answers is not in the document]
Comment (ioaNikas) [reply libs/material/src/lib/delivery/delivery-item/delivery-item.component.html]: 👌👌👌

File: libs/material/src/lib/delivery/delivery-item/delivery-item.component.ts [hunk 16]
```diff
@@ -21,11 +21,13 @@ export class DeliveryItemComponent implements OnInit {
 
   public progression$: Observable<number>;
   public paymentIcon: string;
+  public stakeholders: string;
 
   constructor() {}
 
   ngOnInit() {
     this.paymentIcon = this.delivery.isPaid ? 'payed' : 'not_payed';
+    this.stakeholders = this.delivery.stakeholders.map(({ organization }) => organization.name).join(' | ');
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/delivery-item/delivery-item.component.ts]: It won't update over time ? (if it does, use an observable)

File: libs/material/src/lib/delivery/stakeholder-list/stakeholder-list.component.html [hunk 17]
```diff
@@ -1,15 +1,15 @@
...
       *ngIf="stakeholder.isAccepted"
       [stakeholder]="stakeholder"
     ></delivery-stakeholder-item>
   </div>
-  <button mat-icon-button (click)="navigateToTeamwork()">
-    <mat-icon>add_circle_outline</mat-icon>
-    Add a new signee
-  </button>
+  <div (click)="navigateToTeamwork()" class="add-signee mat-elevation-z6" fxLayout="row" fxLayoutAlign="center center">
+    <mat-icon svgIcon="add"></mat-icon>
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/delivery/stakeholder-list/stakeholder-list.component.html]: Why not using `<mat-icon>add</mat-icon>` instead ?

File: libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.html [hunk 18]
```diff
@@ -35,6 +35,17 @@
           </mat-option>
         </mat-select>
       </mat-form-field>
+      <ng-template #noStep>
+        <mat-form-field appearance="outline" class="mat-select-step">
+          <mat-label>Choose step</mat-label>
+          <mat-select formControlName="stepId">
+            <mat-option value="">
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-form/material-delivery-form.component.html]: If the value is empty, don't add it.

File: libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss [hunk 19]
```diff
@@ -37,28 +51,29 @@ mat-card-content {
...
 
 .border-left {
-  border-left: 1px solid #ECECEC;
+  border-left: 1px solid #ececec;
 }
 
 mat-icon {
-  color:#C1C1C1;
+  color: #c1c1c1;
   float: left;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss]: Never use float. This was already an old deprecated property when I start programing. Use flexbox instead.

File: libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss [hunk 20]
```diff
@@ -37,28 +51,29 @@ mat-card-content {
...
 }
 
 .border-left {
-  border-left: 1px solid #ECECEC;
+  border-left: 1px solid #ececec;
 }
 
 mat-icon {
-  color:#C1C1C1;
+  color: #c1c1c1;
```
Comment (GrandSchtroumpf) [inline libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss]: Is it changing the color of the icon ?

File: libs/material/src/lib/delivery/delivery-item/delivery-item.component.ts (hunk 16 above)
Comment (ioaNikas) [reply libs/material/src/lib/delivery/delivery-item/delivery-item.component.ts]: As delivery is an @Input() from parent already using and observable of deliveries, behavior seems ok. I just checked, new stakeholder is showing immediatly and without needing to reload the page when we add it from another tab/browser.

File: libs/material/src/lib/delivery/stakeholder-list/stakeholder-list.component.html (hunk 17 above)
Comment (ioaNikas) [reply libs/material/src/lib/delivery/stakeholder-list/stakeholder-list.component.html]: This add icon was supposed to be a two-tone icon from ours wireframes. As `add_circle_outline` basic mat-icon is exactly the same, I'll switch to this one until we got the two-tone icon.

File: libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss (hunk 19 above)
Comment (ioaNikas) [reply libs/material/src/lib/material/material-delivery-item/material-delivery-item.component.scss]: Got it. Anyway their is no mat-icon into the html so this is useless css.
Another float is used in notifications-item.scss, but fxFlex seems to behave differently in a mat-list (that has a fixed maximum width). I'll work to change it after Cannes if it's ok.

**Pull request merged**